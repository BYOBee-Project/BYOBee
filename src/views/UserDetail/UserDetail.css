.user-detail {
  width: 65%;
  display: flex;
  flex-direction: column;
  align-items: center;
  border-radius: 5px;
  background-color: rgba(0, 128, 167, 0.384);
  border-radius: 10px;
  margin: 10vh 0 10vh 0;
  padding: 2vh 4vw 2vh 4vw;
}

.user-detail-title {
  font-family: 'Londrina Solid', cursive;
  font-size: 5vh;
  text-align: center;
  padding: 0;
  margin: 6vh;
}

.user-detail-links {
  display: flex;
  flex-direction: column;
  align-items: center;
  margin: 2vh 0 4vh 0;
}

.user-details {
  display: flex;
  flex-direction: column;
  margin: 0;
  width: 60%;
  background-color: rgba(254, 217, 183, 0.195);
  padding: 2vw;
  margin: 2vh 0 4vh 0;
  border-radius: 10px;
}

.user-detail-label {
  font-weight: bold;
}

.user-detail-image {
  width: 40vw;
  margin: 0 0 2vh 0;
  border-radius: 5px;
  filter: drop-shadow(1px 1px 2px #000007);
}

.user-detail-button {
  margin: 2vh 0 4vh 0;
}
