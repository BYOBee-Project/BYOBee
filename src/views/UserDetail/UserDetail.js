import React from 'react';
import './UserDetail.css';
import { findUserBee } from '../../services/submissions';
import { useState, useEffect } from 'react';
import { Link } from 'react-router-dom';

export default function UserDetail(props) {
  const id = props.match.params.id;

  const [userBee, setUserBee] = useState({});
  const [loading, setLoading] = useState(true);

  useEffect(() => {
    const fetchData = async () => {
      const data = await findUserBee(id);
      setUserBee(data);
      setLoading(false);
    };
    fetchData();
  }, [id]);

  if (loading) return <h1 className="loader">Loading...</h1>;
  return (
    <div className="user-detail">
      <h1 className="user-detail-title">{userBee.name}</h1>
      <img className="user-detail-image" src={userBee.image1} alt={userBee.name} />
      <div className="user-details">
        <p className="user-detail-date">
          <span className="user-detail-label">Date Spotted: </span>
          {userBee.date}
        </p>
        <p className="user-detail-location">
          <span className="user-detail-label">Location Spotted: </span>
          {userBee.location}
        </p>
        <p className="user-detail-observation">
          <span className="user-detail-label">Observations: </span>
          {userBee.observations}
        </p>
      </div>
      <button className="user-detail-button">
        <Link to={`/profile`} className="user-detail-link">
          Back to Profile
        </Link>
      </button>
    </div>
  );
}
