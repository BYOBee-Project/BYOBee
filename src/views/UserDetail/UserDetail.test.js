import { render, screen } from '@testing-library/react';
import { MemoryRouter, Route } from 'react-router-dom';
import UserDetail from './UserDetail.js';

test('UserDetail view should render the UserDetail page', async () => {
  const container = render(
    <MemoryRouter>
      <UserDetail
        match={{ params: { id: 32 } }}
        currentUser={{ user: { id: 'd1beb9ec-d137-4e7f-ac5c-4c9197075a15' } }}
      />
    </MemoryRouter>
  );
  await screen.findByText('Date Spotted:');
  expect(container).toMatchSnapshot();
});
