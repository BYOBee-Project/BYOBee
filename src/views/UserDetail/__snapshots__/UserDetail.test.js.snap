// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserDetail view should render the UserDetail page 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="user-detail"
      >
        <h1
          class="user-detail-title"
        >
          Black-notched Bumblebee
        </h1>
        <img
          alt="Black-notched Bumblebee"
          class="user-detail-image"
          src="https://purcoqerkuxhmrkzgmyk.supabase.in/storage/v1/object/public/images/d1beb9ec-d137-4e7f-ac5c-4c9197075a15/3/2022-01-20T20:39:38.457Z.jpg"
        />
        <div
          class="user-details"
        >
          <p
            class="user-detail-date"
          >
            <span
              class="user-detail-label"
            >
              Date Spotted: 
            </span>
            2022-01-13
          </p>
          <p
            class="user-detail-location"
          >
            <span
              class="user-detail-label"
            >
              Location Spotted: 
            </span>
            Portland, ME
          </p>
          <p
            class="user-detail-observation"
          >
            <span
              class="user-detail-label"
            >
              Observations: 
            </span>
            Zach
          </p>
        </div>
        <button
          class="user-detail-button"
        >
          <a
            class="user-detail-link"
            href="/profile"
          >
            Back to Profile
          </a>
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="user-detail"
    >
      <h1
        class="user-detail-title"
      >
        Black-notched Bumblebee
      </h1>
      <img
        alt="Black-notched Bumblebee"
        class="user-detail-image"
        src="https://purcoqerkuxhmrkzgmyk.supabase.in/storage/v1/object/public/images/d1beb9ec-d137-4e7f-ac5c-4c9197075a15/3/2022-01-20T20:39:38.457Z.jpg"
      />
      <div
        class="user-details"
      >
        <p
          class="user-detail-date"
        >
          <span
            class="user-detail-label"
          >
            Date Spotted: 
          </span>
          2022-01-13
        </p>
        <p
          class="user-detail-location"
        >
          <span
            class="user-detail-label"
          >
            Location Spotted: 
          </span>
          Portland, ME
        </p>
        <p
          class="user-detail-observation"
        >
          <span
            class="user-detail-label"
          >
            Observations: 
          </span>
          Zach
        </p>
      </div>
      <button
        class="user-detail-button"
      >
        <a
          class="user-detail-link"
          href="/profile"
        >
          Back to Profile
        </a>
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
